### ISSUE TITLE
Experimental: Incorrect confirmPaymentIntent required params 

### ISSUE THREAD

**Issue by andrepcg** [tipsi/tipsi-stripe#536]
Tagging @fbartho since you developed this part of the code

https://github.com/tipsi/tipsi-stripe/blob/1c8161ce5227b2632c82124e9282b6afbd6db7ff/src/utils/types.js#L273

For the `confirmPaymentIntent` method it's specified in `utils/types` the following:

```
One of the following must be provided:
 * - paymentMethod
 * - paymentMethodId (any supported ID, including IDs for saved card sources)
```

```javascript
export const confirmPaymentIntentPropType = PropTypes.oneOfType([
  PropTypes.shape({
    ...confirmPaymentIntentPropTypeBase,
    paymentMethod: PropTypes.shape(createPaymentMethodPropType).isRequired,
  }),
  PropTypes.shape({
    ...confirmPaymentIntentPropTypeBase,
    paymentMethodId: PropTypes.string.isRequired,
  }),
]).isRequired
```

And this is incorrect since you just need `clientSecret` to use `confirmPayment` (both iOS and Android). Javadocs: https://stripe.dev/stripe-android/com/stripe/android/model/ConfirmPaymentIntentParams.html

These proptypes are also conflicting with the native Tipsi Android code:

https://github.com/tipsi/tipsi-stripe/blob/1c8161ce5227b2632c82124e9282b6afbd6db7ff/android/src/main/java/com/gettipsi/stripe/StripeModule.java#L518

You can clearly see that there's a case for not passing `paymentMethod` nor `paymentMethodId` to `mStripe.confirmPayment()`

**Comment by andrepcg** [comment]
@fbartho already fixed the underlying assertion on iOS. Android is working fine.

https://github.com/tipsi/tipsi-stripe/commit/9ab012f66be5e0ef43aa118e628b9295c756bdc6

**Comment by fbartho** [comment]
oh @andrepcg -- do we still need to make the paymentMethod/paymentMethodId be optional in the PropTypes ?

**Comment by andrepcg** [comment]
> oh @andrepcg -- do we still need to make the paymentMethod/paymentMethodId be optional in the PropTypes ?

If you want to be consistent then yes ;)

**Comment by fbartho** [comment]
Fixed in the 8.0 Release / `experimental` branch
